crates/router/src/db/configs.rs

```diff
@@ -60,6 +65,30 @@ impl ConfigInterface for Store {
             .map_err(Into::into)
             .into_report()
     }
+    async fn find_config_by_key_cached(
+        &self,
+        key: &str,
+    ) -> CustomResult<storage::Config, errors::StorageError> {
+        let redis = &self.redis_conn;
+        let redis_val = redis
+            .get_and_deserialize_key::<storage::Config>(key, "Config")
+            .await;
+        Ok(match redis_val {
+            Err(err) => match err.current_context() {
+                errors::RedisError::NotFound => {
+                    let config = self.find_config_by_key(key).await?;
+                    redis
+                        .serialize_and_set_key(&config.key, &config)
+                        .await
+                        .change_context(errors::StorageError::KVError)?;
+                    config
+                }
+                err => Err(report!(errors::StorageError::KVError)
+                    .attach_printable(format!("Error while fetching config {err}")))?,
+            },
+            Ok(val) => val,
+        })
+    }
```

Comment:
Can we please have this "populate from database if not found in Redis" logic in a single place instead of implementing it for each entity independently?